test/boot/associated_object.rb

```diff
@@ -55,3 +55,33 @@ def great?
 end
 
 Post::Comment.has_object :rating
+
+# Can locate subclasses by grepping for `< Pricing`
+class Pricing < ActiveRecord::AssociatedObject::Polymorphic
+  # Here, `record` will return the post for Post::Pricing objects and the comment for Post::Comment::Pricings.
+
+  def something_conditional
+    execute_behavior if post?
+  end
+end
+
+class Post::Pricing < Pricing
```

Comment:
I was wondering if I should omit having the namespaced `Post::Pricing` classes and try to just make `Pricing` work everywhere. But I think it breaks the whole conceptual model of Associated Objects.

One other thing, I've been considering is to somehow support this:

```ruby
# app/models/pricing.rb
class Pricing < ActiveRecord::AssociatedObject
  class Post::Pricing < self # Post could be loaded here!
  end

  class Comment::Pricing < self
  end
end
```

However, `Post.has_object :pricing` would fail because there's no `post/pricing.rb` file. We could instead try `"Pricing".constantize` which would work, and then skip our Associated Object eager-loading attempt …because it'd be defined right after `Post` finishes loading. I hope this makes sense! All this kind tricky autoloading behavior makes me feel a bit uneasy.

Reply:
This was the first thing that popped into my head. If associated objects are shared, where do they live? It feels like the non-polymorphic approach makes it very clear cut that it's a way of organizing code rather than sharing code. (Whereas concerns, for example, are primarily about sharing code.)